worlds/hk/__init__.py

```diff
@@ -188,7 +189,10 @@ def __init__(self, multiworld, player):
         self.ranges = {}
         self.created_shop_items = 0
         self.vanilla_shop_costs = deepcopy(vanilla_shop_costs)
-        self.grub_count = 0
+
+        # defaulting so completion condition isn't incorrect before pre_fill
+        self.grub_count = 46
+        self.grub_player_count = {player: 46}
```

Comment:
Mentioned in DM's, but you probably want to set this based on the required grub count